app/models/commit_checks.rb

```diff
@@ -0,0 +1,92 @@
+class CommitChecks
+  OUTPUT_TTL = 10.minutes.to_i
+  FINAL_STATUSES = %w(failed error success)
+
+  def initialize(commit)
+    @commit = commit
+  end
+
+  def run
+    self.status = 'running'
+    commands = StackCommands.new(stack)
+    commands.with_temporary_working_directory(commit: commit) do |directory|
+      deploy_spec = DeploySpec::FileSystem.new(directory, stack.environment)
+      Bundler.with_clean_env do
+        capture_all(build_commands(deploy_spec.dependencies_steps, chdir: directory))
+        capture_all(build_commands(deploy_spec.review_checks, chdir: directory))
+      end
+    end
+  rescue Command::Error
+    self.status = 'failed'
+  rescue
+    self.status = 'error'
+    raise
+  else
+    self.status = 'success'
+  end
+
+  def fetch
+    @output ||= begin
+      if redis.set('output', '', ex: OUTPUT_TTL, nx: true)
+        self.status = 'scheduled'
+        PerformCommitChecksJob.perform_later(commit: commit)
+      end
+      redis.get('output').to_s
+    end
+  end
+
+  def status
+    @status ||= redis.get('status')
+  end
+
+  def status=(status)
+    redis.set('status', status, ex: OUTPUT_TTL)
+    @status = status
+  end
+
+  def finished?
+    FINAL_STATUSES.include?(status)
+  end
+
+  def output(since: 0)
+    return fetch if since.zero?
```

Comment:
Doesn't this mean you can call `output` and get nothing back if `since > 0` and the output hasn't been populated yet?